python/tests/test_writer.py

```diff
@@ -310,3 +322,51 @@ def test_writer_fails_on_protocol(existing_table: DeltaTable, sample_data: pa.Ta
     existing_table.protocol = Mock(return_value=ProtocolVersions(1, 2))
     with pytest.raises(DeltaTableProtocolError):
         write_deltalake(existing_table, sample_data, mode="overwrite")
+
+
+def test_writer_with_max_rows(tmp_path: pathlib.Path):
+    def get_multifile_stats(table: DeltaTable) -> Iterable[Dict]:
+        log_path = get_log_path(table)
+
+        # Should only have single add entry
+        for line in open(log_path, "r").readlines():
+            log_entry = json.loads(line)
+
+            if "add" in log_entry:
+                yield json.loads(log_entry["add"]["stats"])
+
+    data = pa.table(
+        {
+            "colA": pa.array(range(0, 1000), pa.int32()),
+            "colB": pa.array([i * random.random() for i in range(0, 1000)], pa.float64())
+        }
+    )
+    path = str(tmp_path)
+    write_deltalake(path, data, file_options=ParquetFileFormat().make_write_options(), max_rows_per_file=100, max_rows_per_group=100)
+
+    table = DeltaTable(path)
+    stats = get_multifile_stats(table)
+
+    assert sum([stat_entry['numRecords'] for stat_entry in stats]) == 1000
+
+
+def test_writer_with_options(tmp_path: pathlib.Path):
+    expected_column_values = [int(datetime(year_, 1, 1, 0, 0, 0).timestamp() * 1e6) for year_ in range(9000, 9010)]
+    data = pa.table(
+        {
+            "colA": pa.array(expected_column_values, pa.timestamp('us'))
+        }
+    )
```

Comment:
You should be able to pass in the raw datetime values into the array constructor:
```suggestion
    expected_column_values = pa.array(
        [datetime(year_, 1, 1, 0, 0, 0) for year_ in range(9000, 9010)], pa.timestamp('us'))
    expected = pa.table({"colA": expected_column_values})
```

Reply:
applied, works nicely :)